### ISSUE TITLE
Gridline Broken

### ISSUE THREAD

**Issue by a7mad7aydar** [vispy/vispy#2363]
Hey guys!

I don't know how to help you reproduce this roblem, but I'll tell you how I got to it.

        self.pointCloudCanvas = vispy.scene.SceneCanvas(keys='interactive', show=True)
        self.pointCloudTab.layout().addWidget(self.pointCloudCanvas.native)
        self.pointCloudView = self.pointCloudCanvas.central_widget.add_view()
        self.pointCloudScatter = visuals.Markers()
        self.pointCloudXYZ =  visuals.XYZAxis(parent=self.pointCloudView.scene)
        self.pointCloudGridLines =  visuals.GridLines(color='w')
        self.pointCloudView.add(self.pointCloudGridLines)
        self.pointCloudScatter.set_data(np.zeros((6, 3)), edge_color=None, size=5) 
        self.pointCloudView.add(self.pointCloudScatter)
        self.pointCloudView.camera = 'turntable'

when zooming or moving the gridlines get really noisy. Is it an error in my code?

![image](https://user-images.githubusercontent.com/29779036/181720046-9d190edc-0ff6-422a-8450-0f38da087db6.png)

**Comment by djhoese** [comment]
You say zooming or moving they look noisy, does that mean they look "good" when zoomed further out? What is the size of your canvas?

**Comment by a7mad7aydar** [comment]
I'm so sorry for the very late answer, I work in my company only in exam-free times.
![image](https://user-images.githubusercontent.com/29779036/190684331-de43f680-8648-44e7-b220-15f09ff24fd3.png)
Yes when zooming out it looks normal (as in the picture). Even without the gridlines, when trying to move closer to a point in the pointcloud, it stops moving closer.


the size of the canvas is responsive and adapts to the screen size.

My colleagues ended up moving back to OpenGL Widget, where it works perfectly fine.

**Comment by djhoese** [comment]
> My colleagues ended up moving back to OpenGL Widget, where it works perfectly fine.

You mean Qt's GL widgets and doing it without vispy?

> Even without the gridlines, when trying to move closer to a point in the pointcloud, it stops moving closer.

This shouldn't be possible unless your point is at the very center of the turntable camera.

Without a minimal example that doesn't require downloading a ton of data, I'm not sure we can debug this further in vispy. My initial guess was that we're dealing with floating point precision in the GPU/vertices. This seams reasonable considering the repeating pattern of the noise in your original image.

**Comment by a7mad7aydar** [comment]
> You mean Qt's GL widgets and doing it without vispy?

Yes, I meant the Qt GL Widget.
> This shouldn't be possible unless your point is at the very center of the turntable camera.

Nope, I'm actually zooming normally inside the pointcloud.
>Without a minimal example that doesn't require downloading a ton of data, I'm not sure we can debug this further in vispy. My initial guess was that we're dealing with floating point precision in the GPU/vertices. This seams reasonable considering the repeating pattern of the noise in your original image.

https://user-images.githubusercontent.com/29779036/191009489-6bcafef6-e490-44c3-a28d-6f0501cb40a8.mp4



Here's a video that shows the problem. Maybe it'll help

**Comment by djhoese** [comment]
Thanks. What kind of transforms do you have on the Visuals in this example? Like what are the min and max coordinates of your MarkersVisual? What `.transform` is being applied to the Visuals or what range/properties are being assigned to the camera to get to this original view?

**Comment by a7mad7aydar** [comment]
Sorry again for the very late answer!

I actually didn't define any transforms at all in my code. I just went with the defaults. The camera is a turntable camera. I tried to play with the initial position of the camera, but it didn't make things better when trying to zoom, so I left it to its default settings too.

**Comment by a7mad7aydar** [comment]
Yes! I got this complain from multiple colleagues, and I develop on Windows and Ubuntu.

**Comment by Jcparkyn** [comment]
I've run into what appears to be the same issue as this. It occurs in two scenarios:
1. Zooming in very far on a point that's not near the scene origin.
2. Viewing the scene in orthographic mode (`fov=0`), with the zoom at or above (approximately) the default. This is much more problematic, as it will occur for almost anyone using an orthographic camera.

Here's a minimal repro, based on https://vispy.org/gallery/scene/grid_layout/grid.html#sphx-glr-gallery-scene-grid-layout-grid-py:

```python
import sys

from vispy import scene, app

canvas = scene.SceneCanvas(keys='interactive')
canvas.size = 600, 600
canvas.show()

grid = canvas.central_widget.add_grid()

b2 = grid.add_view(row=0, col=0)
b2.camera = scene.TurntableCamera(fov=0, azimuth=30, elevation=30)
b2.border_color = (0.5, 0.5, 0.5, 1)

grid2 = scene.visuals.GridLines(parent=b2.scene)

if __name__ == '__main__' and sys.flags.interactive == 0:
    app.run()
```

Screenshot taken without adjusting zoom:
![image](https://github.com/vispy/vispy/assets/51850908/0d919cd2-37d1-44b6-a6c9-30a1a521630c)

And after zooming out:
![image](https://github.com/vispy/vispy/assets/51850908/3a75f8f7-a257-4775-8951-11f58f2655c5)

Environment:
Windows 10, Python 3.9, Vispy 0.12.2, GTX 1060

**Comment by Jcparkyn** [comment]
This also appears to be a duplicate of #1399.
